Arrow Fx Website Doc re-org + improvements. (#2460)
arrow-kt/arrow: arrow-site/docs/docs/fx/README.md

---
layout: docs-fx

> title: Arrow Fx. Typed FP for the masses

Reviewer: ```suggestion
title: Arrow Fx. Async and Concurrent Functional Programming for Kotlin
```

Reviewer: Should probably change other references or look for a better phrase or tagline in this context. FP for the masses sounds empty these days

The library brings purity, referential transparency, and direct imperative syntax to typed FP in Kotlin, and is a fun and easy tool for creating Typed Pure Functional Programs.

If you're not familiar yet with Coroutines in Kotlin, it's recommended to first read Kotlin's [Coroutines Guide](https://kotlinlang.org/docs/coroutines-guide.html) on KotlinX Coroutines.

## Arrow Fx Coroutines Overview 

- Quick Start
  - [Gradle Setup](Gradle-Setup)
  - [Maven Setup](Maven-setup)
- Extensions and data types   

>   - [Parallel combinators](parallel/)

Reviewer: ```suggestion
  - [Parallel](parallel/)
```

>   - [Flow combinators](/apidocs/arrow-fx-coroutines/arrow.fx.coroutines/kotlinx.coroutines.flow.-flow/) 

Reviewer: ```suggestion
  - [Flow](/apidocs/arrow-fx-coroutines/arrow.fx.coroutines/kotlinx.coroutines.flow.-flow/) 
```

  - [Schedule](/apidocs/arrow-fx-coroutines/arrow.fx.coroutines/-schedule/)
  - [Resource](/apidocs/arrow-fx-coroutines/arrow.fx.coroutines/-resource/)
  - [CircuitBreaker](/apidocs/arrow-fx-coroutines/arrow.fx.coroutines/-circuit-breaker/)
  - [Atomic](/apidocs/arrow-fx-coroutines/arrow.fx.coroutines/-atomic/)

>   - [STM](/apidocs/arrow-fx-stm/arrow.fx.stm/-s-t-m/index.html)

Reviewer 2: Is this link correct? All others end in `somelink/` whereas this ends in `somelink/index.html`

Author: Ah ye, the `index.html` can be omitted.
I'll reformat `STM` to be the same as the others.

The STM docs are looking really nice on the webpage 👍
(Arrow Website for next version is broken ATM, we tried to bump Dokka to the new version since the rewrite but we're still blocked :(

Author: ```suggestion
  - [STM](/apidocs/arrow-fx-stm/arrow.fx.stm/-s-t-m/)
```

- [Integrating with 3rd-party libraries](Integrating-with-3rd-party-libraries)

> - Additional Information

Reviewer: These index here is great 👏

    repositories {
        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
    }
}
```

### Gradle BOM file

To avoid specifying the Arrow version for every dependency, a BOM file is available:

```groovy
implementation platform("io.arrow-kt:arrow-stack:$arrow_version")

implementation "io.arrow-kt:arrow-fx-coroutines"
implementation "io.arrow-kt:arrow-fx-stm"
```

## Maven Setup

Add the dependencies that you want to use:
```xml
<dependency>
    <groupId>io.arrow-kt</groupId>
    <artifactId>arrow-core</artifactId>
    <version>0.13.2</version>
</dependency>
```

### Maven BOM file

To avoid specifying the Arrow version for every dependency, a BOM file is available:

```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>io.arrow-kt</groupId>
      <artifactId>arrow-stack</artifactId>
      <version>${arrow.version}</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
```

## Integrating with 3rd-party libraries

Arrow Fx integrates with KotlinX Coroutines Fx, Reactor framework, and any library that can model effectful async/concurrent computations as `suspend`.

> If you are interested in the Arrow Fx library, please contact us in the main [Arrow Gitter](https://gitter.im/arrow-kt/Lobby) or #Arrow channel on the official [Kotlin Lang Slack](https://kotlinlang.slack.com/messages/C5UPMM0A0) with any questions and we'll help you along the way.

Reviewer: ```suggestion
If you are interested in the Arrow Fx library, please contact us in the #Arrow channel on the official [Kotlin Lang Slack](https://kotlinlang.slack.com/messages/C5UPMM0A0) with any questions and we'll help you along the way.
```
We should consider dropping gitter from all places as it just fragments our community which is active in Slack.

Author: Yes, 100% agreed. Trying to gather everyone in Slack would be best. I have an easy time keeping my eyes on Slack, but multiple channels that I don't actively use is cumbersome.
